core/src/test/scala/com/microsoft/azure/synapse/ml/nbtest/SynapseUtilities.scala

```diff
@@ -230,31 +222,156 @@ object SynapseUtilities extends HasHttpClient {
     }
   }
 
-  def getSynapseToken: String = {
-    val spnKey: String = Secrets.SynapseSpnKey
+  private def getBigDataPoolBicepPayload(bigDataPoolName: String,
+                                         poolLocation: String,
+                                         poolNodeSize: String,
+                                         createdAtTs: String): String = {
+    s"""
+    |{
+    |  "name": "${bigDataPoolName}",
+    |  "location": "${poolLocation}",
+    |  "tags": {
+    |    "createdBy": "SynapseE2E Tests",
+    |    "createdAt": "${createdAtTs}"
+    |  },
+    |  "properties": {
+    |    "autoPause": {
+    |      "delayInMinutes": "10",
+    |      "enabled": "true"
+    |    },
+    |    "autoScale": {
+    |      "enabled": "true",
+    |      "maxNodeCount": "10",
+    |      "minNodeCount": "3"
+    |    },
+    |    "cacheSize": "20",
+    |    "dynamicExecutorAllocation": {
+    |      "enabled": "true",
+    |      "maxExecutors": "8",
+    |      "minExecutors": "2"
+    |    },
+    |    "isComputeIsolationEnabled": "false",
+    |    "nodeCount": "0",
+    |    "nodeSize": "${poolNodeSize}",
+    |    "nodeSizeFamily": "MemoryOptimized",
+    |    "provisioningState": "Succeeded",
+    |    "sessionLevelPackagesEnabled": "true",
+    |    "sparkVersion": "3.2"
+    |  }
+    |}
+    |""".stripMargin
+  }
 
-    val uri: String = s"https://login.microsoftonline.com/$TenantId/oauth2/token"
+  def createSparkPools(poolCount: Int = 1,
+                       subscriptionId: String,
+                       resourceGroupName: String,
+                       workspaceName: String): Array[String] = {
+    val sparkPool = new Array[String](poolCount)
```

Comment:
this is still state that you seem to be mutating down below. consiuder turning your .foreach into a map and returning the spark pool id